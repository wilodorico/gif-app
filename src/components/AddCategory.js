import React, { useState } from 'react'
import PropTypes from "prop-types"

const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
      e.preventDefault()
      if(inputValue.trim().length > 2){
         setCategory(cat => [inputValue, ...cat])
         setInputValue("")
      }
  }

  return (
    <>
      <h2>{inputValue}</h2>
      <form onSubmit={handleSubmit}>
        <input type='text' value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  )
}

AddCategory.propTypes = {
   setCategory: PropTypes.func.isRequired
}

export default AddCategory
