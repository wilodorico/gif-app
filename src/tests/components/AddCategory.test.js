import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('Test component <AddCategory />', () => {

  const setCategory = jest.fn()
  let wrapper = shallow(<AddCategory setCategory={setCategory} />)

  beforeEach(() => {
     jest.clearAllMocks()
     wrapper = shallow(<AddCategory setCategory={setCategory} />)
  })

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change the text box', () => {
    const input = wrapper.find('input')
    const value = 'Hello world'
    input.simulate('change', { target: { value },
    })
  })

  test('should not send the form if not value', () => { 

   wrapper.find('form').simulate('submit', { preventDefault(){} })

   expect(setCategory).not.toHaveBeenCalled()

   })

   test('should setCategory and clear the textbox', () => { 
      const value = "Hello world"

      wrapper.find('input').simulate('change', {target: { value }})

      wrapper.find('form').simulate('submit', {preventDefault(){} })

      expect(setCategory).toHaveBeenCalled();
      expect(setCategory).toHaveBeenCalledTimes(1);
      expect(setCategory).toHaveBeenCalledWith( expect.any(Function) );

      expect(wrapper.find('input').prop('value')).toBe('')

    })


})
