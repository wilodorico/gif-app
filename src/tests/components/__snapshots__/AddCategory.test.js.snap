// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <AddCategory /> should display correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
